src/adera/Machines/Rocket.cpp

```diff
@@ -74,21 +76,64 @@ std::vector<WireOutput*> MachineRocket::existing_outputs()
     return {};
 }
 
-SysMachineRocket::SysMachineRocket(ActiveScene &rScene)
-    : SysMachine<SysMachineRocket, MachineRocket>(rScene)
-    , m_updatePhysics(rScene.get_update_order(), "mach_rocket", "controls", "physics",
-        [this](ActiveScene& rScene) { this->update_physics(rScene); })
+void SysMachineRocket::add_functions(ActiveScene &rScene)
 {
-
+    rScene.debug_update_add(rScene.get_update_order(), "mach_rocket", "controls", "physics",
+                            &SysMachineRocket::update_physics);
+    rScene.debug_update_add(rScene.get_update_order(), "mach_rocket_construct", "vehicle_activate", "vehicle_modification",
+                            &SysMachineRocket::update_construct);
 }
 
 //void SysMachineRocket::update_sensor()
 //{
 //}
 
+void SysMachineRocket::update_construct(ActiveScene& rScene)
+{
+    auto view = rScene.get_registry()
```

Comment:
So this isn't incorrect, but I would point out that frequently people tend to assume that an undecorated "auto" without a &, is a value, and not a reference.

IMHO if you want people to assume this is a reference (which I think it is, based on context?) you'd want to put the &.

Reply:
I don't know the details of how EnTT works, but I think the view is a very lightweight object that is constructed on-demand, so this really should be a value.